### DISCUSSION TITLE
Postal v2

### DISCUSSION THREAD

**Discussion by adamcooke** [postalserver/postal#1513] [Announcements]
![68747470733a2f2f73686172652e6164616d2e61632f32312f417274626f6172642d73456a354e4548483637447038555045504b2e706e67](https://github.com/postalserver/postal/assets/4765/c1716292-83a0-4996-a2f5-e38826d104ba)

Over the last few days, we've been busy at work preparing a new future for Postal. The direction we are now taking will allow us to more easily continue to support the platform and ensure things are as easy to use as possible. Here's everything that's happened:

* **The only supported way to install Postal is now using containers.** Checking out the repository to run the code is not supported any longer. If you are already running Postal, [read this guide](https://docs.postalserver.io/install/upgrade-to-v2) to find out how to upgrade your installation.

* **Click/open tracking has been simplified.** This means that Postal will no longer attempt to issue SSL certificates and serve content directly. This has now been moved into the main Postal web server and you'll need to install and manage your own web proxy to handle SSL termination.

* **User management has been simplified.** Previously, in order to add users to organizations, you needed to invite them by e-mail. This was something left over from when Postal was a SaaS product. This has now been removed. Admin users can now add and manage users through the web interface. This also means a single non-admin user can be provided with access to multiple organizations.

* **We have a new documentation site.** The old GitHub wiki-based documentation has been replaced with a lovely new site at [docs.postalserver.io](https://docs.postalserver.io). This has been set up by @jaydrogers and looks great. The content on here has been reviewed and is now inline with the new container-based distribution system.

* **IP address priorities have been added.** If you use IP pools, you can now use them to manually warm up new IP addresses. This allows you to send a small volume of email through certain IPs.  You just need to set a priority on the IP addresses. 

* **We have a new logo!** It's yellow and it's P shaped. 

* **A number of bug fixes.** A number of bugs have been fixed and minor enhancement introduced. 

* **GitHub Discussions is now the main place to request help & support.** Previously, our GitHub issues was overrun with users asking for help and support to use and install Postal. Unfortunately, this means that finding actionable development tasks was becoming harder and harder. We have now opened up GitHub Discussions and encourage everyone to use them when asking for support or suggesting new features. If you create an **issue** for something that isn't a bug, it will simply be closed.

* **I've recorded a new introduction video.** You can [watch it on YouTube](https://youtu.be/d1Lzw_Q_fJQ). 

* **You can grab us on Slack.** We have a Slack channel on the Krystal Open Source Slack workspace. [Join us](https://slack.k.io) if you want to - you might get an answer to your question slightly more quickly.

Having written all that, there's still lots to do and we're hopeful we'll be able to look at this over the next few months. Here's a few of things on the roadmap:

* **Upgrades to many underlying components.** We need to upgrade Rails as well as a number of other dependencies.

* **Resolve the remaining issues.** At the time of writing there are still 94 issues on GitHub. These need to be reviewed, triaged and either actioned or closed. 

* **Expansion of the test suite.** We have a fair amount of work to do to expand the test coverage in the application. This is very much something that needs some love.

* **New HTTP API.** The current API provides a very small number of features. We have a new API framework that we will be open sourcing in the next few months and we want to use it to provide a far more feature-rich HTTP API. Our new framework generate [beautiful documentation](https://developers.katapult.io/api) and would be an excellent addition to Postal.

* **Support for additional spam checking services.** We currently support SpamAssassin but I would like to see us have support for rspamd to improve spam detection in the application.

Phew. That's a fair number of words - thanks for getting this far. I know we (the core team) have been quite quiet for the last couple of years but we're still heavy users of Postal ourselves and are very much hoping to dedicated more time to the project going forward. If there's any Ruby developers who would like a project to contribute to, please join us on Slack, we'd love to have a chat about things.

**Comment by jabaco** [comment]
Congratulations! this looks very promising and will become the # 1 open source email server (if not already)

**Comment by cmmata** [comment]
Congrats! I'm supper happy to see that this project is gaining traction again! It's the best open source mail server in my opinion, and I was quite sad thinking about that it could be abandoned or something like that.

**Comment by donatid** [comment]
Congratulations!! I have been waiting for this moment while I'm using version 1.x and it's amazing.
Anyway, I hope that manual install will be supported in the future, old fashioned people like me love the "manual way" :)).

Once again, thanks for the effort on this awesome project!

**Comment by ryanbutterfield** [comment]
Congrats on v2! 

We installed postal for the first time a couple of days ago using your new container installation and documentation. All self explanatory and no major issues, even installing it on rootless docker. 

Loving the application, it's working really well, and we can finally say goodbye to postmark. Looking forward to your new API so that we can integrate with suppressions.

Thanks adam and team for all your work.

**Comment by nkgarg21** [comment]
Hello

Can anybody help with https://github.com/postalserver/postal/discussions/2046  ?

**Comment by Pilpin** [comment]
Hello,

> New HTTP API. The current API provides a very small number of features. We have a new API framework that we will be open sourcing in the next few months and we want to use it to provide a far more feature-rich HTTP API. Our new framework generate [beautiful documentation](https://developers.katapult.io/api) and would be an excellent addition to Postal.

Are you still planning on open sourcing the new http api framework ?

Thanks

**Comment by 365stealer** [comment]
Hello,

I would appreciate some assistance with my VPS setup. At the moment, I’m unable to access the VPS management dashboard or configure the PTR/rDNS for the VPS IP.

Could you please confirm whether Postal can be installed and its dashboard accessed before PTR/rDNS is configured?

Also, the domain I intend to use with Postal is registered for one year but is not currently hosted on a web hosting account. Will Postal work with the domain in this situation, provided the necessary DNS records are properly configured?

Thank you for your assistance.

**Reply by AlPhaSh3ll** [reply]
I can help you to check your server and postal configuration also your registrar of your server but please make sure ptr/rdns is enable because even if we get a chance to install postal without ptr/rdns your email will he rejected by 90% of esp
